wtxmgr/tx.go

```diff
@@ -3460,7 +3464,15 @@ func (s *Store) Balance(ns, addrmgrNs walletdb.ReadBucket, minConf int32,
 	_, syncHeight := s.MainChainTip(ns)
 	switch balanceType {
 	case BFBalanceFullScan:
-		return s.balanceFullScan(ns, addrmgrNs, minConf, syncHeight, all, account)
+		bals, err := s.balanceFullScan(ns, addrmgrNs, minConf, syncHeight)
+		if err != nil {
+			return 0, err
+		}
+		bal, ok := bals[account]
+		if !ok {
```

Comment:
might want to add a comment here saying that if no results were found for the account, therefore it must have a zero balance